### PR TITLE
[Communication] - Phone Numbers - Changed User-Agent in request

### PR DESCRIPTION
**Pull request by jbeauregardb**

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py [hunk 1]
```diff
@@ -25,14 +25,14 @@ def process_response(self, response):
                     if isinstance(item, str):
                         body["phoneNumbers"] = [self._replacement]
                         break
-                    elif "phoneNumber" in item:
+                    if "phoneNumber" in item:
                         item['phoneNumber'] = self._replacement
-                    elif "id" in item:
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py:30]: why we change from elif to if here?

File: sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py [hunk 2]
```diff
@@ -25,14 +25,14 @@ def process_response(self, response):
                     if isinstance(item, str):
                         body["phoneNumbers"] = [self._replacement]
                         break
-                    elif "phoneNumber" in item:
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py:28]: why we change from elif to if here?

File: sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py [hunk 3]
```diff
@@ -25,14 +25,14 @@ def process_response(self, response):
...
+                    if "phoneNumber" in item:
                         item['phoneNumber'] = self._replacement
-                    elif "id" in item:
+                    if "id" in item:
                         item['id'] = self._replacement
             response['body']['string'] = json.dumps(body)
             response['url'] = self._replacement
             return response
-        except (KeyError, ValueError):
+        except (KeyError, ValueError, TypeError):
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py:35]: why we need this change for the User-agent change?

**Review by sacheu — COMMENTED** [review]
i left a few comments.

File: sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py (hunk 3 above)
Comment (jbeauregardb) [reply sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py:35]: This is a fix for the type error issue we have in the search available phone numbers test. We weren't catching that exception before. It's totally unrelated to the user agent issue, though.

File: sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py (hunk 2 above)
Comment (jbeauregardb) [reply sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py:28]: I wanted to sanitize both values. Having an elif only sanitized the phone number or the id, but not both at the same time.

**Review by sacheu — APPROVED** [review]
Thanks. Looks good.

**Pull request merged**